### PR TITLE
read trove.yml in current dir

### LINKED ISSUE #115: suggest: read trove.yml in the current directory (or load with option)
Many projects use Makefile, gradle, and npm and more to configure commands for their project, but we often need to read the code inside because those tools are difficult to write a detailed description.

If `hoard` have the ability to read `trove.yml` in the current directory, we will be able to search for those commands and give them descriptions.

This makes it possible to add trove.yml to each repository to manage commands.

```
# project root
Makefile # declare command
trove.yml # search, and execute in Makefile command.
```

---

Whether to load automatically or optionally needs to be discussed.

### PR DESCRIPTION
**Pull request by kyoto7250**
close #115 

This PR adds an option in config to read trove.yml in the current directory.

Thank you in advance.

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -149,9 +152,19 @@ fn load_or_build(path: &Path) -> Result<HoardConfig, Error> {
...
+            is_config_dirty = true;
+        }
         if is_config_dirty {
             save_config(&loaded_config, &hoard_config_path)?;
         }
+
+        let path_buf = Path::new(HOARD_FILE).to_path_buf();
+        if loaded_config.read_from_current_directory.unwrap() && path_buf.exists() {
+            loaded_config.trove_home_path = Some(path_buf);
+        }
```
Comment (kyoto7250) [inline src/config.rs:166]: Added here for commands such as `hoard new`.

File: src/config.rs [hunk 2]
```diff
@@ -42,6 +43,7 @@ impl HoardConfig {
             tertiary_color: Some(Self::default_colors(2)),
             command_color: Some(Self::default_colors(3)),
             parameter_token: Some(Self::default_parameter_token()),
+            read_from_current_directory: Some(false)
```
Comment (Hyde46) [inline src/config.rs:46]: Create the default value for `read_from_current_directory` similarly to the other default values
`read_from_current_directory: Some(Self::default_read_from_current_dir()`

Then, add
`HoardConfig::default_read_from_current_dir()`
Which just returns `false`

Might seem a bit overkill, but I planned to abstract this part out at some point

Comment (Hyde46) [reply src/config.rs:46]: Also, good choice to add the `Option` for the bool, to support older config files :smile:

File: src/config.rs (hunk 1 above)
Comment (Hyde46) [reply src/config.rs:166]: Now that the loaded trove in `HoardConfig` can be a other trove file other than the one in its home path, we could refactor the name form `trove_home_path` to just `trove_path`

**Review by Hyde46 — CHANGES_REQUESTED** [review]
Looks great so far! :rocket: 
Some smaller change requests before mergin this in :smile:

**Comment by kyoto7250** [comment]
@Hyde46 
Thank you for your PR review!
I fixed my PR, so please review again :)

**Comment by Hyde46** [comment]
Awesome!
Checking out your changes.

On a separate note, thank you for all the work on a bunch of issues! :rocket: 
I'd like to hook you up with a couple of stickers if you're interested. Hit me up at denis.heid68@googlemail.com with your address and i'll send you some, wherever you're located at.
stickers

**Review by Hyde46 — APPROVED** [review]
Looks great, thanks for your contribution!
:+1:

**Pull request merged**